pages/profile.tsx

```diff
@@ -1,26 +1,35 @@
 import { useUser } from '@auth0/nextjs-auth0/client'
 import styles from '@/styles/pageStyles/Profile.module.css'
 import dynamic from 'next/dynamic'
+import { use, useEffect, useState } from 'react'
+import { grabDataToDisplay } from '@/helperFunctions/grabDataToDisplay'
+import { testTrips } from '@/constants/test'
 
 function profilePage() {
   const NavBar = dynamic(() => import('@/components/NavBar'))
   const ProfileComponent = dynamic(() => import('@/components/Profilepage/ProfileComponent'))
   const ProfileDefault = dynamic(() => import('@/components/Profilepage/ProfileDefault'))
-  const { user } = useUser()
-  const userId: number = user?.['https://example.com/id'] as number
-  const profilepicture = user?.picture
-  //TODO Check local storage OR get trips for length
-  const trips: any = ['1', '2', '3'] 
-  
+  const { user, isLoading } = useUser()
+  const userId: number | any = process.env.NEXT_PUBLIC_AUTH0_USER_ID
+    ? user?.[process.env.NEXT_PUBLIC_AUTH0_USER_ID]
+    : null
+
+  const [tripsNumber, setTripsNumber] = useState<number>(0)
+
+  useEffect(() => {
+    if (!!user) {
+      grabDataToDisplay(userId, 'trips').then((val) => {
+        if (Object.keys(val).length > 0) {
```

Comment:
```suggestion
        if (Object.keys(val)?.length > 0) {
```
This could just be me being paranoid, but when I see dot notation it worries me in case the object COULD be null. Same with line23